tests/test_json_schema.py

```diff
@@ -5681,6 +5681,37 @@ class ModelTest(BaseModel):
     assert calls == 1
 
 
+def test_mixed_type_callable_json_schema_extra():
+    def add_key1(s):
+        s['key_extra1'] = 'extra1'
+
+    def add_key2(s):
+        s['key_extra2'] = 'extra2'
+
+    CustomField = Annotated[
+        int,
+        Field(json_schema_extra={'key11': 'value11'}),
+        Field(json_schema_extra=add_key1),
+        Field(json_schema_extra={'key1': 'value1'}),
+        Field(json_schema_extra=add_key2),
+    ]
+
+    class Model(BaseModel):
+        a: CustomField
+
+    field_model_json_schema = Model.model_json_schema()['properties']['a']
+    field_model_json_schema.pop('title')
```

Comment:
Maybe don't do this `pop` until we're going to compare with the type adapter's json schema, and add a comment explaining why the `pop` happens (type adapter doesn't have field names, so `title` doesn't make sense in context).